db/write_batch.cc

```diff
@@ -405,9 +405,11 @@ Status WriteBatch::Iterate(Handler* handler) const {
   Status s;
   char tag = 0;
   uint32_t column_family = 0;  // default
-  while ((s.ok() || UNLIKELY(s.IsTryAgain())) && !input.empty() &&
+  bool last_was_try_again = false;
+  while (((s.ok() && !input.empty()) || UNLIKELY(s.IsTryAgain())) &&
```

Comment:
why should it TryAgain in case input is empty?

Reply:
When reaching the last key/value in the batch, if that key is a duplicate then the first attempt of handling it will return TryAgain. The look condition however breaks since the input is now empty (after reading the last key/value) so it will not be tried again. And yes, we should try again even if there is nothing left in the input.